File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/aceData.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

export enum AceCardSize {
    Medium = "Medium",
    Large = "Large"
}

/**
 * Sharepoint Ace Data object
 */
export class AceData {
    private cardSize: AceCardSize;
    private dataVersion: string;
    private id: string;
    private title: string;
    private iconProperty: string;
    /**
     * Initializes a new instance of the AceData class
     */
    public AceData(){
        // Do nothing
    }

    /**
     * Sets card size property of type AceCardSize
     */
    public set CardSize(cardSize: AceCardSize){
        this.cardSize = cardSize;
    }

    /**
     * Gets card size property of type AceCardSize
     */
    public get CardSize(): AceCardSize {
        return this.cardSize;
    }

    /**
     * Sets data version property of type string
     */
    public set DataVersion(dataVersion: string){
        this.dataVersion = dataVersion;
    }

    /**
     * Gets data version property of type string
     */
    public get DataVersion(): string {
        return this.dataVersion; 
    }

    /**
     * Sets id property of type string
     */
    public set Id(id: string) {
        this.id = id;
    }

    /**
     * Gets id property of type string
     */
    public get Id(): string {
        return this.id;
    }

    /**
     * Sets title property of type string
     */
    public set Title(title: string){
        this.title = title;
    }

    /**
     * Gets title property of type string
     */
    public get Title(): string {
        return this.title;
    }

    /**
     * Sets icon property of type string
     */
    public set IconProperty(iconProperty: string){
        this.iconProperty = iconProperty;
    }

    /**
     * Gets icon property of type string
     */
    public get IconProperty(): string{
        return this.iconProperty;
    }
}

File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/AceData.cs
﻿// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Text;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// SharePoint Ace Data object.
    /// </summary>
    public class AceData
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="AceData"/> class.
        /// </summary>
        public AceData()
        {
            // Do nothing
        }

        /// <summary>
        /// This enum contains the different types of card templates available in the SPFx framework.
        /// </summary>
        public enum AceCardSize
        {
            /// <summary>
            /// Small
            /// </summary>
            Small,

            /// <summary>
            /// Medium
            /// </summary>
            Medium,

            /// <summary>
            /// Large
            /// </summary>
            Large
        }

        /// <summary>
        /// Gets or Sets the card size of the adaptive card extension of type <see cref="AceCardSize"/> enum.
        /// </summary>
        /// <value>This value is the size of the adaptive card extension.</value>
        [JsonProperty(PropertyName = "cardSize")]
        [JsonConverter(typeof(StringEnumConverter))]
        public AceCardSize CardSize { get; set; }

        /// <summary>
        /// Gets or Sets the version of the data of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the version of the adaptive card extension.</value>
        [JsonProperty(PropertyName = "dataVersion")]
        public string DataVersion { get; set; }

        /// <summary>
        /// Gets or Sets the id of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the ID of the adaptive card extension.</value>
        [JsonProperty(PropertyName = "id")]
        public string Id { get; set; }

        /// <summary>
        /// Gets or Sets the title of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the title of the adaptive card extension.</value>
        [JsonProperty(PropertyName = "title")]
        public string Title { get; set; }

        /// <summary>
        /// Gets or Sets the icon property of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the icon of the adaptive card extension.</value>
        [JsonProperty(PropertyName = "iconProperty")]
        public string IconProperty { get; set; }

        /// <summary>
        /// Gets or Sets the login uri of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the icon of the adaptive card extension.</value>
        [JsonProperty(PropertyName = "uri")]
        public Uri SignInUri { get; set; }

        /// <summary>
        /// Gets or Sets the connection name of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the icon of the adaptive card extension.</value>
        [JsonProperty(PropertyName = "connectionName")]
        public string ConnectionName { get; set; }
    }
}
